src/pynxtools/dataconverter/helpers.py

```diff
@@ -699,18 +656,14 @@ def is_valid_data_field(value, nxdl_type, path):
     accepted_types = NEXUS_TO_PYTHON_DATA_TYPES[nxdl_type]
     output_value = value
 
+    if isinstance(value, list):
+        return all(is_valid_data_field(v, nxdl_type, path) for v in value), output_value
+
```

Comment:
This produces one warning per element in a list, if the datatype is not correct. @RubelMozumder had a different solution for this in #554.